javaslang/src/main/java/javaslang/Value.java

```diff
@@ -901,7 +901,9 @@ default CharSeq toCharSeq() {
     @SuppressWarnings("unchecked")
     default SortedSet<T> toSortedSet() throws ClassCastException {
         final Comparator<T> comparator;
-        if (this instanceof SortedSet<?>) {
+        if (this instanceof TreeSet<?>) {
+            return (TreeSet<T>) this;
```

Comment:
Yes, but currently the toSortedSet() method's javadoc states that a TreeSet is returned. In the case we already have a TreeSet we are able to return the same instance. If we have a SortedSet, we need to create a new TreeSet instance...

This is really comparator-hell, addressed in #1488 and #1495.